org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/XMLSchemaErrorCode.java

```diff
@@ -176,37 +176,35 @@ public static Range toLSPRange(XMLLocator location, XMLSchemaErrorCode code, Obj
 			}
 			return XMLPositionUtility.selectTrimmedText(offset, document);
 		}
-		case SchemaLocation:
+		case SchemaLocation: { //xml xsi:schemaLocation
+			SchemaLocation schemaLocation = document.getSchemaLocation();
+			DOMRange locationRange = schemaLocation.getAttr().getNodeAttrValue();
+			return locationRange != null ? XMLPositionUtility.createRange(locationRange) : null;
+		}
 		case schema_reference_4: {
-			DOMRange locationRange = null;
-			if (code.equals(SchemaLocation)) {
-				SchemaLocation schemaLocation = document.getSchemaLocation();
-				locationRange = schemaLocation.getAttr().getNodeAttrValue();
-			} else {
-				String hrefLocation = arguments.length == 1 ? (String) arguments[0] : null;
-				// Check if location comes from a xml-model/@href
-				locationRange = XMLModelUtils.getHrefNode(document, hrefLocation);
-				if (locationRange == null) {
-					NoNamespaceSchemaLocation noNamespaceSchemaLocation = document.getNoNamespaceSchemaLocation();
-					if (noNamespaceSchemaLocation != null) {
-						locationRange = noNamespaceSchemaLocation.getAttr().getNodeAttrValue();
-					} else {
-						SchemaLocation schemaLocation = document.getSchemaLocation();
-						if (schemaLocation != null) {
-							String invalidSchemaPath = arguments[0] instanceof String ? (String) arguments[0] : null;
+			String hrefLocation = arguments.length == 1 ? (String) arguments[0] : null;

```

Comment:
Add a comment which explains that this error code comes from when XSD fiile path is not valid with 3 usecases : 

 * case 1: <?xml-model href="bad.xsd"
 * case 2 : xsi:schemaLocation="http://foo bad.xsd"
 * case 3: xsi:noNamespaceSchemaLocation="bad.xsd"